py3status/i3status.py

```diff
@@ -522,6 +522,14 @@ def suspend_i3status(self):
 
     @profile
     def run(self):
+        # if the i3status process dies we want to restart it.
+        while self.lock.is_set():
+            self.spawn_i3status()
+            # check if we never worked properly and if so quit now
+            if not self.ready:
+                break
+
```

Comment:
we may want to throttle this a bit or add a maximal respawn count in cause this ends badly over and over